frontend/lib/src/components/shared/StreamlitMarkdown/styled-components.ts

```diff
@@ -188,6 +188,10 @@ export const StyledStreamlitMarkdown =
           marginRight: theme.spacing.none,
         },
 
+        code: {
```

Comment:
Is this only relevant for inline code or also code blocks. In `CodeBlock` we also have styled classes related to inline code and code blocks, which I believe are used in Markdown. I think it might be better applied in these Styled classes instead of here: https://github.com/streamlit/streamlit/blob/e87a933eb900fc0ef2a6463f135ef5d14ad819fe/frontend/lib/src/components/elements/CodeBlock/styled-components.ts#L27

Reply:
thanks for the comment!
I moved styling to the `StyledInlineCode`, and it solves the reported issue with Markdown. 

For the `StyledCode` component, we already specify `whiteSpace: "pre", and I don't want to change it, since this could be desired behaviour for general st.code other components etc. 

```python
st.code("""(['zero', ' one', '    two'], 3)""")
```
also displays whitespaces correctly